packages/flutter_tools/test/integration.shard/test_utils.dart

```diff
@@ -165,18 +162,27 @@ class ProcessResultMatcher extends Matcher {
   @override
   bool matches(dynamic item, Map<dynamic, dynamic> matchState) {
     final ProcessResult result = item as ProcessResult;
-
-    if (stdoutSubstring != null) {
-      _foundStdout = (result.stdout as String).contains(stdoutSubstring!);
-      matchState['stdout'] = result.stdout;
+    bool foundStdout = true;
+    bool foundStderr = true;
+
+    final String stdout = result.stdout as String;
+    final String stderr = result.stderr as String;
```

Comment:
No, this will always be either `List<int>` (if you chose not to encode it) or `String` https://github.com/dart-lang/sdk/blob/main/sdk/lib/io/process.dart#L525